instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/snippet/ServletOutputStreamInjectionHelper.java

```diff
@@ -62,7 +67,12 @@ public static boolean handleWrite(InjectionState state, ServletOutputStream out,
     state.setAlreadyInjected(); // set before write to avoid recursive loop
     out.write(b);
     try {
-      byte[] snippetBytes = SnippetHolder.getSnippetBytes(state.getCharacterEncoding());
+      if (state.getWrapper().isNotSafeToInject()) {
+        // content length already set and sent, don't inject
+        return false;
+      }
+      byte[] snippetBytes = ExperimentalSnippetHolder.getSnippetBytes(state.getCharacterEncoding());
+      state.getWrapper().updateContentLengthIfPreviouslySet();
       out.write(snippetBytes);
```

Comment:
I think I'd move these lines outside (before) the try block

```
      if (state.getWrapper().isNotSafeToInject()) {
        // content length already set and sent, don't inject
        return false;
      }
      state.getWrapper().updateContentLengthIfPreviouslySet();
```

Reply:
updateContentLengthIfPreviouslySet could not be moved outside the try catch as it needs to be set only if the encoding successfully performed